Sources/AsyncMoya/MoyaProvider+Async.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+import Moya
+
+#if swift(>=5.5)
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+public extension MoyaProvider {
+    /// Async request
+    /// - Parameter target: Entity, with provides Moya.Target protocol
+    /// - Returns: Result type with response and error
+    func asyncRequest(_ target: Target) async -> Result<Response, MoyaError> {
```

Comment:
Just wondering why we chose for `asyncRequest(_:)` here, and not `request(_:)`? :)

Reply:
rename method